File: Tutoriels/PHP/z1_Make_Ecommerce_Site_Weekend/php/class/GProcess.php
<?php   
    abstract class GProcess {
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            $lKey = GConfig::Instance()->getData("PROCESS");
            if($lKey == "PRINT") return GProcessPrint::Instance();
            if($lKey == "CONFIG") return GProcessConfig::Instance();
            if($lKey == "ECOMMERCE") return GProcessEcommerce::Instance();
            if($lKey == "SIGNUP") return GProcessSignup::Instance();
            return GProcessPrint::Instance();
        }
        //===============================================
        abstract public function run();
        //===============================================
    }
?>

File: Tutoriels/Java/Projets/code/GProject/src/manager/GProcess.java
//===============================================
package manager;
//===============================================
abstract public class GProcess {
	//===============================================
	public static GProcess Instance() {
		String lKey = GConfig.Instance().getData("PROCESS");
		if(lKey.equals("PRINT")) return GProcessPrint.Instance();
		if(lKey.equals("CONFIG")) return GProcessConfig.Instance();
		return GProcessPrint.Instance();
	}
	//===============================================
	abstract public void run();
	//===============================================
}
//===============================================
